src/reply.ml

```diff
@@ -252,8 +252,10 @@ module Decoder = struct
         else
           let k code' decoder = go code code' [ Bytes.sub_string raw_crlf off (len - 2) ] decoder in
           number k decoder
-      | Some chr ->
-        leave_with decoder (`Unexpected_char chr)
+      | Some _ ->
+        let raw_crlf, off, len = peek_while_eol decoder in
+        let reply = v code [ Bytes.sub_string raw_crlf off (len - 2) ] in
+        decoder.pos <- decoder.pos + len ; k reply decoder
```

Comment:
This would allow any garbage with after the code (unless the protocol expects something specific). Maybe we should simply check if there is just space missing and we receive `\r\n`.

I played a bit with the code and maybe we could have something like:
```
        let raw_crlf, off, len = peek_while_eol decoder in
        if chr == '\r' && (Bytes.get raw_crlf (off+1)) == '\n'
        then
          let reply = v code [ Bytes.sub_string raw_crlf off (len - 2) ] in
          decoder.pos <- decoder.pos + len ; k reply decoder
        else
          leave_with decoder (`Unexpected_char chr)
```
I suppose my code is a bit ugly way of doing it, but you probably know cleaner way. Like I said, I'm still learning ocaml.